### PR TITLE
Update the Kubernetes agent system info

### PR DESCRIPTION
**Pull request by richma-ms**
Update the Kubernetes runtime system info to return the following:
OS = "Kubernetes"
version = K8s API version
arch = list of architectures in the cluster (and the node count for each)

### CODE REVIEW

File: edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/KubernetesRuntimeInfoProvider.cs [hunk 1]
```diff
@@ -110,31 +110,55 @@ public async Task<Stream> GetModuleLogs(string module, bool follow, Option<int>
...
+
+            // Use "Kubernetes" and the current API version as analogy to device OS/version
             string osType = "Kubernetes";
-            string arch = "Kubernetes";
-            string version = "Kubernetes";
-            return Task.FromResult(new SystemInfo(osType, arch, version));
+            string version = k8sNodes.ApiVersion;
+
+            // Enumerate each node and count the unique architectures
+            var archList = new List<(string arch, int count)>();
```
Comment (dmolokanov) [inline edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/KubernetesRuntimeInfoProvider.cs]: I think it's a very C/C++ way of doing this stuff. In a C# we have a very powerful way to work with collections called LINQ. 
This code can be re-written in a more declarative way:
```
		var achitectures = k8sNodes.Items
			.Where(node => node?.Status?.NodeInfo != null)
			.Select(node => node.Status.Status.NodeInfo.Architecture)
			.GroupBy(architecture => architecture)
			.Select(architectures => $"{architectures.Key} ({architectures.Count})";
		
		var arch = string.Join("\t", achitectures);
		return new SystemInfo(osType, arch, version);
```

File: edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/KubernetesRuntimeInfoProvider.cs [hunk 2]
```diff
@@ -110,31 +110,55 @@ public async Task<Stream> GetModuleLogs(string module, bool follow, Option<int>
...
-            // }
+            V1NodeList k8sNodes = await this.client.ListNodeAsync();
+            string accumulatedArch = string.Empty;
+
+            // Use "Kubernetes" and the current API version as analogy to device OS/version
             string osType = "Kubernetes";
-            string arch = "Kubernetes";
-            string version = "Kubernetes";
-            return Task.FromResult(new SystemInfo(osType, arch, version));
+            string version = k8sNodes.ApiVersion;
```
Comment (dmolokanov) [inline edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/KubernetesRuntimeInfoProvider.cs]: I think we need to take a version of k8s api server by making http request to `/version` api. I believe the c# client should have this method implemented

File: edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/KubernetesRuntimeInfoProvider.cs [hunk 3]
```diff
@@ -110,31 +110,55 @@ public async Task<Stream> GetModuleLogs(string module, bool follow, Option<int>
...
+                {
+                    Events.NullNodeInfoResponse(node?.Metadata?.Name ?? "UNKNOWN");
+                    break;
+                }
+            }
+
+            // Construct the accumulated architecture string
+            foreach (var (arch, count) in archList)
+            {
+                accumulatedArch += arch + "(" + count + ")\t";
```
Comment (dmolokanov) [inline edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/KubernetesRuntimeInfoProvider.cs]: @darobs Should we preserve the same structure as Docker mode or we can make it different? For example like this: 
```
{
  OperatingSystemType: "Kubernetes",
  Version: "1.12",
  Architectures: [
    { Name: "x86", NodesCount: 10 },
    { Name: "arm64", NodesCount: 1}
  ]
}
```

**Comment by richma-ms** [comment]
Abandoning this PR as listing nodes requires a ClusterRole, and we don't want the agent to have a ClusterRole.  Going to try a different approach.

**Pull request closed without merging**